config.go

```diff
@@ -1034,7 +1034,13 @@ func normalizeAddresses(addrs []string, defaultPort string) []string {
 	seen := map[string]struct{}{}
 	for _, addr := range addrs {
 		if _, _, err := net.SplitHostPort(addr); err != nil {
-			addr = net.JoinHostPort(addr, defaultPort)
+			// If the address is an integer, then we assume it is *only* a
+			// port and default to binding to that port on localhost
+			if _, err := strconv.Atoi(addr); err == nil {
```

Comment:
In a future change, we may want to differentiate between RPC/REST listeners and P2P listeners. As is, this change would listen on localhost with the specified port even for the P2P listener, which may not be ideal as P2P listens on all interfaces by default.